core/main/src/broker/rules_engine.rs

```diff
@@ -202,6 +208,12 @@ impl RuleEngine {
         }
         None
     }
+    pub fn get_rule_by_method(&self, method: &str) -> Option<Rule> {
+        if let Some(rule) = self.rules.rules.get(&method.to_lowercase()).cloned() {
+            return Some(rule);
+        }
+        None
+    }
```

Comment:
Nitpick: This could just be:

`
pub fn get_rule_by_method(&self, method: &str) -> Option<Rule> {
    self.rules.rules.get(&method.to_lowercase()).cloned()
}
`

Reply:
Agreed. Will change as suggested.